server.js

```diff
@@ -0,0 +1,73 @@
+var express = require('express');
+
+import { request, abort, } from 'request-promise';
+
+const port = process.env.PORT || 1337;
+
+const app = express();
+
+async function incoming2outgoingPayload(payload) {
+  import jsontoxml from 'jsontoxml';
+  const itemsToXML = {}
+  for(let i = 0; i <= payload.items.length; i++) {
+    const startDate = new Date(payload.items[i].start);
+    const endDate = new Date(payload.items[i].end);
+    itemsToXML[`vacation-${startDate.toString()}-${endDate.toString()}`] = payload.items[i].approved == true ? true : false;
+  }
+  return jsontoxml(itemsToXML)
+}
+
+async function fetch(url, payload) {
+  console.log("Triggering url " + url);
+
+  body = incoming2outgoingPayload(payload);
+
+  return request({ url: url, method: 'Post', body: body })
+}
+
+async function cancel(request) {
+  return await abort(request);
+}
```

Comment:
Is this function being used? I don't see it called anywhere. If not, or even if you were planning to use it later, probably makes sense to leave it out for now.

You'd be able to clean up the code a bit by removing this function and the import for `abort` above, as well.